pkg/scheduler/cache/cache.go

```diff
@@ -992,14 +992,26 @@ func (sc *SchedulerCache) taskUnschedulable(task *schedulingapi.TaskInfo, reason
 		Message: message,
 	}
 
-	if podConditionHaveUpdate(&pod.Status, condition) {
+	updateCond := podConditionHaveUpdate(&pod.Status, condition)
+	updateNomiNode := podNominatedNodeNameNeedUpdate(&pod.Status, nominatedNodeName)
+
+	if updateCond || updateNomiNode {
 		pod = pod.DeepCopy()
 
+		if updateCond && podutil.UpdatePodCondition(&pod.Status, condition) {
+			klog.V(3).Infof("Updating pod condition for %s/%s to (%s==%s)", pod.Namespace, pod.Name, condition.Type, condition.Status)
+		}
+
+		if updateNomiNode {
```

Comment:
Better add a comment here to explain why this is added？

Reply:
done